app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/signatures/GetSignaturesViewModel.kt

```diff
@@ -128,19 +128,19 @@ class GetSignaturesViewModel @Inject constructor(
                 _state.update { state ->
                     state.copy(
                         showContentForFactorSource = State.ShowContentForFactorSource.Device,
-                        isRetryButtonEnabled = true // enable the Retry button, reason is at line 184
+                        isRetryButtonEnabled = true // enable the Retry button, reason is at line 185
                     )
                 }
                 sendEvent(event = Event.RequestBiometricToAccessDeviceFactorSources)
             }
 
             null -> {
+                sendEvent(event = Event.AccessingFactorSourceCompleted)
                 accessFactorSourcesIOHandler.setOutput(
                     output = AccessFactorSourcesOutput.EntitiesWithSignatures(
                         signersWithSignatures = state.value.entitiesWithSignatures
                     )
                 )
-                sendEvent(event = Event.AccessingFactorSourceCompleted)
             }
```

Comment:
changed the order of the event here, otherwise it causes issues with the navigation (because of the bottom sheet presented on top of the stack)